### PR TITLE
[Merged by Bors] - feat: Product of injective functions on sets

### PR DESCRIPTION
**Pull request by YaelDillies**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Set/Function.lean [hunk 1]
```diff
@@ -1853,6 +1861,43 @@ protected lemma InvOn.extendDomain (h : InvOn g₁ g₂ s t) :
 
 end Set
 
+namespace Set
+variable {α₁ α₂ β₁ β₂ : Type*} {s₁ : Set α₁} {s₂ : Set α₂} {t₁ : Set β₁} {t₂ : Set β₂}
+  {f₁ : α₁ → β₁} {f₂ : α₂ → β₂} {g₁ : β₁ → α₁} {g₂ : β₂ → α₂}
+
+lemma InjOn.prodMap (h₁ : s₁.InjOn f₁) (h₂ : s₂.InjOn f₂) :
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Set/Function.lean:1868]: I find dot notation for InjOn doesn't really read well, but won't block on it

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

File: Mathlib/Data/Set/Function.lean (hunk 1 above)
Comment (urkud) [reply Mathlib/Data/Set/Function.lean:1868]: Why not use `Prod.map` in the statements?

File: Mathlib/Data/Set/Function.lean [hunk 2]
```diff
@@ -1041,6 +1041,14 @@ theorem BijOn.image_eq (h : BijOn f s t) : f '' s = t :=
   h.surjOn.image_eq_of_mapsTo h.mapsTo
 #align set.bij_on.image_eq Set.BijOn.image_eq
 
+lemma BijOn.forall {p : β → Prop} (hf : BijOn f s t) : (∀ b ∈ t, p b) ↔ ∀ a ∈ s, p (f a) where
+  mp h a ha := h _ $ hf.mapsTo ha
+  mpr h b hb := by obtain ⟨a, ha, rfl⟩ := hf.surjOn hb; exact h _ ha
```
Comment (urkud) [inline Mathlib/Data/Set/Function.lean:1046]: Alternative proof: `by rw [← hf.image_eq, forall_mem_image]`

File: Mathlib/Data/Set/Function.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Data/Set/Function.lean:1868]: because it gets simplified to its definition as `Prod_map` (not a typo!!) is a simp lemma

**Comment by urkud** [comment]
Thanks! :tada:
bors merge

**Pull request closed without merging**